### ISSUE TITLE
Run the matched carrier-alignment and training-data scaling experiment

### ISSUE THREAD

**Issue by Sino-Huang** [Sino-Huang/NovPhy#63]
## Category

Experiment.

## Scientific questions

Issue #57 cannot distinguish inadequate training coverage from deployment-carrier mismatch. Run a matched factorial experiment that answers both questions before any new final gameplay benchmark is generated:

1. Does increasing independent multi-shot training lineages improve the continuous world model when the deployment carrier is held fixed?
2. Does training on the deployment-aligned carrier improve over source-derived carrier training when lineage coverage is held fixed?

## Frozen experiment matrix

Use the same continuous predictor architecture, optimizer-example budget, hyperparameters, and prospectively frozen training seeds in a 2×2 primary matrix:

| Carrier | Training coverage |
| --- | --- |
| source-derived diagnostic carrier | six-lineage subset |
| source-derived diagnostic carrier | full successor training role |
| deployment-aligned carrier from #60 | same six-lineage subset |
| deployment-aligned carrier from #60 | same full successor training role |

The issue-15 checkpoint is retained as an external historical reference, not silently substituted for a matched cell. Use at least three frozen training seeds, or a larger replicate count justified and frozen from calibration-only variance before model-selection scoring.

## Primary estimands

- Complete-rollout recursive prediction error/AUC for continuous h1 and h15.
- Realized action-ranking regret on identical frozen legal candidate sets at held-out decision states.
- Action-ranking sensitivity across distinct level states rather than seed-only action repetition.
- Model execution/nonfinite failures and declared physical diagnostics.
- Training and inference compute.

Local teacher-forced error is diagnostic only and cannot override recursive or action-ranking results.

## Decision and candidate freeze

Before opening model-selection results, use training/calibration roles to freeze practical-effect thresholds, uncertainty procedures, multiplicity handling, failed-run treatment, and the candidate advancement rule.

The data-scale hypothesis is supported only if the full deployment-carrier model improves over the six-lineage deployment-carrier model by the frozen practical margins without a declared physical-diagnostic regression. Carrier alignment is assessed by the matched source-versus-deployment contrasts at each scale.

If no full deployment-carrier candidate passes, publish `not_supported_by_this_experiment` and stop the final-gameplay chain. Do not collect or open a new final gameplay set merely to search for a favorable outcome. If a candidate passes, freeze its predictor/controller checkpoint and exact deployment configuration for #64 and the final gameplay experiment.

## Acceptance criteria

- All four matched cells run the same frozen architecture, optimizer-example exposure, lineage manifests, training seeds, and evaluation states.
- Training/calibration/model-selection lineages remain disjoint and complete; no frame-level split or final gameplay access occurs.
- Results report per-seed and aggregate local, recursive, action-ranking, physical-diagnostic, failure, and compute metrics with uncertainty.
- Fixed continuous-h1 and continuous-h15 evaluations are reported separately.
- The analysis distinguishes the data-coverage main effect, carrier main effect, and their interaction without treating the historical checkpoint as a matched replicate.
- The exact decision is `supported` or `not_supported_by_this_experiment` under the prospectively frozen rule.
- A supported result produces one frozen deployment-aligned predictor/controller configuration; an unsupported result produces no outcome-conditioned retraining.
- Machine-readable evidence, tables, plots, rerun commands, and exact validation are published with visible progress logs.

## Non-goals

- Symbolic-mode or adaptive-granularity claims.
- Final gameplay-success evaluation.
- Architecture search during the matched primary experiment.
- Expert actions, target frames, or human demonstrations.

## Blocked by

- #60 — Implement a deployment-aligned temporal carrier for multi-shot gameplay dynamics.
- #61 — Implement lineage-scaled continuous-model retraining and matched gameplay ablations.
- #62 — Generate a role-isolated multi-shot gameplay-dynamics successor cohort.

**Comment by Sino-Huang** [comment]
Implementation is ready on `env` at `a88cb22fa35b50c2e6e5bf082a4b0a0051125f9c`; the long experiment remains delegated and this issue stays open.

Delivered:
- real issue-62 fixed-step h1/h15 source and deployment carrier materialization over complete multi-shot lineages, with shot-segment-aware recursion
- frozen nested 6/200/1000/3000 training ladder, both carriers, three seeds, equal 4,000,000-example optimizer exposure, and 24 total learning-curve cells (12 primary)
- batched visual parsing for the 1,710,554 issue-62 frames with visible per-role progress
- 24 distinct calibration plus 24 distinct model-selection ranking states, each with the observed action and four prospectively frozen legal perturbations; Unity replays retain failures at worst cost
- strict calibration/model-selection exposure order: model-selection ranking collection and scoring are rejected until the calibration-only decision rule is frozen
- separate local diagnostic, recursive h1/h15 AUC, normalized carrier contrasts, realized ranking regret/sensitivity, physical/failure, uncertainty, training/inference compute, interaction, resource-limited saturation, tables, SVG plots, supported/not-supported decision, compact publication, and exact validation
- model-selection remains non-final; issue-57 final gameplay and all fresh final gameplay remain unopened

Verification completed without writes or new candidate replays:
- real-data dry run passed on 8 training + 1 calibration + 1 model-selection issue-62 lineages
- all 12 matched primary cells trained in the dry run
- recursive h1/h15 and five-candidate recursive ranking paths passed
- 49 focused issue-60/61/62/63 tests passed

Run from the repository after activating `novphy` and sourcing `env.sh`, in this exact order. Every command is unbuffered and logs progress:

```
python -u -m scripts.run_issue_63_matched_experiment --prepare 2>&1 | tee -a data/issue-63-prepare.log
python -u -m scripts.run_issue_63_matched_experiment --collect-calibration-ranking --start-display 2>&1 | tee -a data/issue-63-calibration-ranking.log
python -u -m scripts.run_issue_63_matched_experiment --train 2>&1 | tee -a data/issue-63-train.log
python -u -m scripts.run_issue_63_matched_experiment --score-calibration 2>&1 | tee -a data/issue-63-calibration.log
python -u -m scripts.run_issue_63_matched_experiment --freeze-decision 2>&1 | tee -a data/issue-63-freeze.log
python -u -m scripts.run_issue_63_matched_experiment --collect-model-selection-ranking --start-display 2>&1 | tee -a data/issue-63-model-selection-ranking.log
python -u -m scripts.run_issue_63_matched_experiment --score-model-selection 2>&1 | tee -a data/issue-63-model-selection.log
python -u -m scripts.run_issue_63_matched_experiment --publish 2>&1 | tee -a data/issue-63-publish.log
python -u -m scripts.run_issue_63_matched_experiment --validate 2>&1 | tee -a data/issue-63-validate.log
```

Do not run model-selection ranking or scoring before `--freeze-decision`. If the exact result is unsupported, the workflow publishes `not_supported_by_this_experiment` and produces no deployment configuration for #64.

**Comment by Sino-Huang** [comment]
Preparation schema fix pushed at `4abcee6ffd8e719d946caba7cdb4b8b027723524`.

Root cause: the persisted issue-62 v4 plan stores exact declarations under `nested_training_scales`; the issue-63 runner incorrectly read the derived CLI-summary name `nested_training_scale_counts`. The fix reads the v4 declarations and uses their frozen `slot_identities` for the 6/200/1000/3000 memberships.

Verification:
- regression test failed before and passes after the fix
- exact v4 artifact resolves `(6, 200, 1000, 3000)` with strictly nested memberships
- 26 focused issue-60/61/63 tests pass
- the exact `--prepare` command now passes the former boundary and begins normal role/parser loading
- the bounded verification wrote no partial issue-63 artifacts

Safe to rerun the original progress-logged command after pulling `env`.

**Comment by Sino-Huang** [comment]
Ranking replay path correction pushed at `fbc419949b085b60e393600988c2ebc2add75836`.

Root cause: the literal state/candidate identities were nested into the runtime slot identity, and `_install_level` used that slot as the XML filename. The resulting filename exceeded the filesystem component limit and every perturbed candidate failed before Unity execution with `ENAMETOOLONG`.

Correction:
- runtime slot and branch artifact paths now use compact project-standard SHA-256 identities
- the 44 existing failures are preserved under their original paths and disclosed by `issue-63-ranking-infrastructure-correction-v1:sha256:3e95113e685dea4e4f275239f7715327e0a765cbbb8b22e527e094ff2fb412fb`
- all 44 occurred at `pre_execution_level_install`, opened no candidate outcome, and are superseded/replayed rather than assigned worst cost
- publication and exact validation bind the correction identity
- an imminent misplaced summary-identity block that would have broken `--freeze-decision` was also corrected

Verification:
- 28 focused tests pass
- prior filename measured 416 bytes; compact installed name is within the 255-byte limit
- one real displayed replay passed: engine startup, 241 aligned frames, `stable_entered`, exact trajectory validation
- one completed compact trajectory is retained and will be skipped on resume; zero compact-path failures

Safe resume command:

```
python -u -m scripts.run_issue_63_matched_experiment --collect-calibration-ranking --start-display 2>&1 | tee -a data/issue-63-calibration-ranking.log
```

**Comment by Sino-Huang** [comment]
Superseding the SHA-based runtime-path correction from the preceding comment at the operator request. Commit `cc2c1ee1622fec7470994821ffc62b947d1e3b8e` uses no hash/checksum for ranking runtime paths or correction identity.

Replacement naming is direct ordinal mapping:
- branch directory: `rank-cal-s001-c01`
- Unity slot: `issue-63-rank-cal-s001-c01-slot-v3`
- correction identity: `issue-63-ranking-infrastructure-correction-v2:pre-execution-enametoolong`

The prior SHA-path verification trajectory and v1 correction record are preserved only under `ranking-collection/verification-probes/` and are excluded from experiment outcomes. The 44 original ENAMETOOLONG records remain preserved and are superseded by the v2 ordinal-path correction; no candidate outcome had been opened.

Verification: 28 focused tests pass, no SHA/hash helper remains in the ranking runner, and a real displayed ordinal replay completed with 241 aligned frames and `stable_entered`. The retained `rank-cal-s001-c01` trajectory will be skipped on resume.

**Comment by Sino-Huang** [comment]
Scoring performance correction is pushed in 053629a (env).

- Calibration and model-selection scoring now use 4 bounded worker threads by default; --score-workers can override it.
- Evaluation carrier/ranking bundles stay shared in memory, while each worker owns its model.
- Score artifacts use compact ordinal v2 identities and .score-v2.json paths. No new SHA-256 integrity work is performed.
- The scorer no longer rereads each approximately 1 GB training report just to copy wall time; it records the immutable relative report reference.
- Existing v1 score files are preserved for audit and ignored on resume.
- Load, checkpoint, cell, and completion progress is emitted with line buffering.

A bounded live calibration start reached parallel workers=4 and launched cells 1-4 together. During evaluation the process rose above one core (about 150% CPU in the sample) with GPU activity. Tests: issue-63 8/8; shared lineage scaling 12/12.

Resume command:
python -u -m scripts.run_issue_63_matched_experiment --score-calibration --score-workers 4 2>&1 | tee -a data/issue-63-calibration.log

**Comment by Sino-Huang** [comment]
Publication schema correction is pushed in e2d6637 (env).

Root cause: compact score v2 replaced embedded training_wall_seconds with training_report_reference, but the v1 CSV publisher still indexed the removed field. The failed attempt wrote a 2.9 GB recursive-identity results file and a header-only CSV before raising KeyError.

Correction:
- publication reads numeric training wall time from only the canonical tail of each referenced training report
- compact ordinal results-v2 and summary-v2 identities use no new SHA/checksum work
- v2 JSON/text outputs are atomic and idempotent, so publication can safely resume after interruption
- validation consumes the v2 evidence and no longer parses each multi-gigabyte training report in full
- failed v1 partial artifacts remain preserved and are excluded by the v2 paths

Verification:
- exact --publish completed twice consecutively
- decision: not_supported_by_this_experiment
- model-selection cells: 24
- final evaluation: unopened
- results-v2.json: 3.9 KB; summary-v2: 828 bytes; per-seed table: 24 data rows
- 22 focused issue-63/shared lineage-scaling tests pass

**Comment by Sino-Huang** [comment]
Issue #63 is complete and exactly validated.

Final experiment outcome:
- decision: not_supported_by_this_experiment
- all learning-curve cells: 24
- matched primary cells: 12
- calibration score cells validated: 24
- model-selection score cells validated: 24
- selected deployment configuration: none
- final evaluation opened: false

The exact validator passed all training/calibration/model-selection carrier bundles, all 24 checkpoints, both score inventories, compact v2 results/summary evidence, and publication references. Under the prospectively frozen rule, the unsupported result correctly stops the final-gameplay advancement chain without outcome-conditioned retraining.

Final implementation correction: e2d66377c76c1a889b2471f8ef62bea44235a529 on env.

**Comment by Sino-Huang** [comment]
## Corrective successor plan after the exact unsupported result

Issue #63's published result remains `not_supported_by_this_experiment`; reopening does not overwrite, rerun, or reinterpret that evidence. The diagnosis found:

- full deployment training passed the recursive-h15 improvement gate but failed recursive h1 and action-ranking gates;
- local deployment prediction was small while self-fed recursive predictions diverged badly;
- the five-action ranking probe was outcome-uninformative in 14/24 states and distinguished pig removal in only 7/24;
- the full 3,000-lineage cell received about 2.5 effective epochs under the fixed 4M-example budget and the final rung remained non-saturated;
- there were no nonfinite or execution failures.

Corrective work is split by responsibility:

- #66 — implement broad outcome-independent ranking probes and a three-checkpoint h15 deterministic ensemble/PETS-lite scorer;
- #67 — implement short-unrolled self-conditioned h15 training and absolute rollout-stability accounting;
- #68 — generate a fresh non-final calibration/model-selection holdout with broader candidate outcomes and audit videos;
- #69 — run the corrected stability and uncertainty-aware advancement experiment and freeze one exact candidate only if it passes.

Canonical dependency graph:

`#63 → (#66 ∥ #67); #66 → #68; (#67 + #68) → #69`

Conditional downstream branch:

- if #69 is `supported`: `#69 → #64 → #65 → #35 → #36 → #37 → #38`;
- if #69 is unsupported: do not run #64/#65; hand the bounded negative result and unopened-final audit directly to #35, then continue `#35 → #36 → #37 → #38`.

No #64 final benchmark may be generated merely because these implementation tickets complete. It requires the exact supported disposition from #69.

**Comment by Sino-Huang** [comment]
The reopened #63 diagnosis and successor handoff are complete.

The original matched experiment remains exactly validated with the immutable decision `not_supported_by_this_experiment`, 24/24 learning-curve cells, 24/24 calibration scores, 24/24 model-selection scores, no selected deployment configuration, and final evaluation unopened.

The reopened investigation identified the bounded reasons for non-advancement and published the corrective path in #66–#69. No #63 experiment was rerun: its model-selection role is already exposed, so changing training, candidates, thresholds, or analysis and rerunning under this ticket would be outcome-conditioned. No repository artifact or prior conclusion was changed.

Next executable ticket: #66. #67 may proceed in parallel; #68 and #69 preserve the fresh-holdout boundary before any possible #64 advancement.
